strawberry_django_plus/integrations/guardian.py

```diff
@@ -1,20 +1,27 @@
 import contextlib
 import dataclasses
 import weakref
-from typing import Optional, Union, cast
+from typing import TYPE_CHECKING, Optional, Union, cast
 
-from django.contrib.auth import get_user_model
-from django.contrib.auth.models import Group
+from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from guardian import backends as _guardian_backends
 from guardian.conf import settings as guardian_settings
 from guardian.core import ObjectPermissionChecker as _ObjectPermissionChecker
 from guardian.models.models import GroupObjectPermissionBase, UserObjectPermissionBase
 from guardian.utils import get_anonymous_user as _get_anonymous_user
 from guardian.utils import get_group_obj_perms_model, get_user_obj_perms_model
+from typing_extensions import TypeAlias
 
 from strawberry_django_plus.utils.typing import UserType
 
+if TYPE_CHECKING:
+    from django.contrib.auth.models import Group
+else:
+    Group = None
+UserOrGroup: TypeAlias = Union[Group, UserType]
```

Comment:
In this case you are only using `Group` for type checking, so this should be enough

```suggestion

UserOrGroup: TypeAlias = Union["Group", UserType]
```